libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -460,8 +461,19 @@ void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, ui
         }
         uint32_t* count_ptr = (uint32_t*) args;
         (*count_ptr)++;
-        GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received geometry_msgs/TwistStamped");
-        // TODO implement the velocity control to AP
+        auto *external_control = AP::externalcontrol();
+        if (external_control == nullptr) {
+            break;
+        }
+        /*
+          get commands from ENU to NED frame
+         */
+        Vector3f linear_velocity {
+            float(rx_velocity_control_topic.twist.linear.y),
+            float(rx_velocity_control_topic.twist.linear.x),
+            float(-rx_velocity_control_topic.twist.linear.z) };
+        const float yaw_rate = -rx_velocity_control_topic.twist.angular.z;
+        external_control->set_linear_velocity_and_yaw_rate(linear_velocity, yaw_rate);
```

Comment:
Here are the test results for the frame orientations:
Positive x (correctly went East):
![x_pos](https://github.com/ArduPilot/ardupilot/assets/62964137/ab72d359-0c12-47e1-aea0-a2464bffaf3d)
Positive y (correnctly went North):
![y_pos](https://github.com/ArduPilot/ardupilot/assets/62964137/6b5d4681-3723-4e2f-bba4-4f5cc8f49968)
Negative z (Wrongfully went Up):
![z_negative](https://github.com/ArduPilot/ardupilot/assets/62964137/c8a3003a-aa5c-4152-be47-239e861c301a)
Positive Yaw (correctly went CounterClockwise looking from above):
![yaw_pos](https://github.com/ArduPilot/ardupilot/assets/62964137/a838572c-dfab-4dc3-aefe-3b6b7e07edb5)

Thus, simply changing the linear z axis should solve it.
```suggestion
        Vector3f linear_velocity {
            float(rx_velocity_control_topic.twist.linear.y),
            float(rx_velocity_control_topic.twist.linear.x),
            float(rx_velocity_control_topic.twist.linear.z) };
        const float yaw_rate = -rx_velocity_control_topic.twist.angular.z;
        external_control->set_linear_velocity_and_yaw_rate(linear_velocity, yaw_rate);
```

Reply:
Also tested to see if the frame orientations were consistent with #23541. Everything except the z linear axis is perfect

Reply:
> Here are the test results for the frame orientations: Positive x (correctly went East): ![x_pos](https://user-images.githubusercontent.com/62964137/259555182-ab72d359-0c12-47e1-aea0-a2464bffaf3d.gif) Positive y (correnctly went North): ![y_pos](https://user-images.githubusercontent.com/62964137/259555211-6b5d4681-3723-4e2f-bba4-4f5cc8f49968.gif) Negative z (Wrongfully went Up): ![z_negative](https://user-images.githubusercontent.com/62964137/259555219-c8a3003a-aa5c-4152-be47-239e861c301a.gif) Positive Yaw (correctly went CounterClockwise looking from above): ![yaw_pos](https://user-images.githubusercontent.com/62964137/259555241-a838572c-dfab-4dc3-aefe-3b6b7e07edb5.gif)
> 
> Thus, simply changing the linear z axis should solve it.

This negative for Z should go in the copter API. The external AHRS said positive Z is down, and copter implements it differently.

Reply:
> This negative for Z should go in the copter API. The external AHRS said positive Z is down, and copter implements it differently.

Yes, we should definatly be using NED where positive Z is down. It can get a bit mixed up because positive altitude is up and I have not converted the position controller to NED yet but I have done most of the work in the background to make that happen. I intend it to be done by the time we fix the units.

Reply:
That would solve it but break other vehicles. I have added a negative in the Copter implementation in this commit, as well as a comment why.
b4251a1ad46269b4796fdca07291c5bbbb5eb5db

Reply:
@pedro-fuoco In my testing, I see positive X going west. 
![image](https://github.com/ArduPilot/ardupilot/assets/25047695/5857c792-203a-4201-9bfd-11439b1ea1c8)